server/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/random/RandomSamplerAggregatorFactory.java

```diff
@@ -87,11 +87,10 @@ private Weight getWeight() throws IOException {
                 shardSeed == null ? context.shardRandomSeed() : shardSeed
             );
             BooleanQuery booleanQuery = new BooleanQuery.Builder().add(query, BooleanClause.Occur.FILTER)
-                .add(context.query(), BooleanClause.Occur.FILTER)
+                .add(context.query(), BooleanClause.Occur.MUST)
                 .build();
-            weight = context.searcher().createWeight(context.searcher().rewrite(booleanQuery), ScoreMode.COMPLETE_NO_SCORES, 1f);
+            weight = context.searcher().createWeight(context.searcher().rewrite(booleanQuery), ScoreMode.COMPLETE, 1f);
         }
         return weight;
     }
-
```

Comment:
I still need to write a test for this, but I like a preliminary review before I do so.